src/test/java/org/kohsuke/github/GHCheckRunBuilderTest.java

```diff
@@ -0,0 +1,88 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.kohsuke.github;
+
+import org.junit.Test;
+
+import java.util.Date;
+
+@SuppressWarnings("deprecation") // preview
+public class GHCheckRunBuilderTest extends AbstractGitHubWireMockTest {
+
+    @Test
+    public void createCheckRun() throws Exception {
+        GHCheckRun checkRun = gitHub.getRepository("jglick/github-api-test")
+                .createCheckRun("foo", "4a929d464a2fae7ee899ce603250f7dab304bc4b")
+                .withStatus(GHCheckRun.Status.COMPLETED)
+                .withConclusion(GHCheckRun.Conclusion.SUCCESS)
+                .withDetailsURL("http://nowhere.net/stuff")
+                .withExternalID("whatever")
+                .withStartedAt(new Date(999_999_000))
+                .withCompletedAt(new Date(999_999_999))
+                .withOutput("Some Title", "what happened…")
+                .withAnnotation("stuff.txt", 1, GHCheckRun.AnnotationLevel.NOTICE, "hello to you too")
+                .withTitle("Look here")
+                .done()
+                .withImage("Unikitty", "https://i.pinimg.com/474x/9e/65/c0/9e65c0972294f1e10f648c9780a79fab.jpg")
+                .withCaption("Princess Unikitty")
+                .done()
+                .done()
+                .withAction("Help", "what I need help with", "doit")
+                .create();
+        assertEquals("completed", checkRun.getStatus());
+        assertEquals(1, checkRun.getOutput().getAnnotationsCount());
+        assertEquals(535049329, checkRun.id);
+    }
+
+    @Test
+    public void createCheckRunManyAnnotations() throws Exception {
+        GHCheckRunBuilder.DraftOutput output = gitHub.getRepository("jglick/github-api-test")
+                .createCheckRun("big", "4a929d464a2fae7ee899ce603250f7dab304bc4b")
+                .withConclusion(GHCheckRun.Conclusion.SUCCESS)
+                .withOutput("Big Run", "Lots of stuff here »");
+        for (int i = 0; i < 101; i++) {
+            output.withAnnotation("stuff.txt", 1, GHCheckRun.AnnotationLevel.NOTICE, "hello #" + i).done();
+        }
+        GHCheckRun checkRun = output.done().create();
+        assertEquals("completed", checkRun.getStatus());
+        assertEquals("Big Run", checkRun.getOutput().getTitle());
+        assertEquals(101, checkRun.getOutput().getAnnotationsCount());
+        assertEquals(537614832, checkRun.id);
+    }
+
+    @Test
+    public void createCheckRunNoAnnotations() throws Exception {
+        GHCheckRun checkRun = gitHub.getRepository("jglick/github-api-test")
+                .createCheckRun("quick", "4a929d464a2fae7ee899ce603250f7dab304bc4b")
+                .withConclusion(GHCheckRun.Conclusion.NEUTRAL)
```

Comment:
Conclusion is optional when `competedAt` is not set.  need test for that.

Reply:
You mean when `status` is not `completed`? Again, these are just aspects of the GitHub semantics which are uninteresting to test here since they are not exercising novel and subtle code idioms.

Reply:
@jglick 
I'm looking at https://developer.github.com/v3/checks/runs/#parameters

> Required if you provide completed_at or a status of completed. 

May not be super interesting but we should verify this not providing conclusion works.  If you remove this line from this test, we get that. 

Frankly, we should also have a test verifying the error returned when conclusion is not set but `status` is `completed`.  The test framework makes adding tests pretty quick.

Reply:
> we should verify this not providing conclusion works

Sure.

> If you remove this line from this test

Yeah, so long as the code coverage report does not then start complaining that there is insufficient coverage percentage for some nested class, as it was yelling at me before…

> we should also have a test verifying the error returned when…

Can do that. As a general observation, GitHub does not document error codes. Even when I have raised support requests and specifically complained that documentation neglects to specify the error code for some corner case, they have _rejected_ the request saying that their general guidelines mention error responses (when in fact those guidelines are so vague that they could easily be interpreted in several ways).